engine/class_modules/monk/sc_monk.cpp

```diff
@@ -4306,6 +4480,186 @@ struct yulon_spell_t : public monk_spell_t
     p()->pets.yulon.spawn( p()->talent.mistweaver.invoke_yulon_the_jade_serpent->duration(), 1 );
 
     p()->buff.invokers_delight->trigger();
+
+    p()->buff.courage_of_the_white_tiger->trigger();
+  }
+};
+
+// ==========================================================================
+// Unity Within
+// ==========================================================================
+
+struct unity_within_t : public monk_spell_t
+{
+  unity_within_t( monk_t *p, util::string_view options_str )
+    : monk_spell_t( p, "unity_within", p->talent.conduit_of_the_celestials.unity_within )
+  {
+    parse_options( options_str );
+
+    harmful = false;
+    usable_while_casting = true;
+  }
+
+  bool ready() override
+  {
+    return p()->buff.unity_within->up();
+  }
+
+  void execute() override
+  {
+    // Expiring this before the execute so that there is an easy way to apply the damage bonus.
+    p()->buff.unity_within->expire();
+
+    monk_spell_t::execute();
+  }
+};
+
+// ==========================================================================
+// Celestial Conduit
+// ==========================================================================
+
+// This is a separate spell from the normal Flight of the Red Crane damage spell
+struct flight_of_the_red_crane_celestial_dmg_t : public monk_spell_t
+{
+  flight_of_the_red_crane_celestial_dmg_t( monk_t *p )
+    : monk_spell_t( p, "flight_of_the_red_crane_celestial_dmg",
+                    p->talent.conduit_of_the_celestials.flight_of_the_red_crane_celestial_dmg )
+  {
+    background = true;
+    aoe        = p->talent.conduit_of_the_celestials.flight_of_the_red_crane->effectN( 1 ).base_value();
+  }
+
+  double composite_aoe_multiplier( const action_state_t *state ) const override
+  {
+    double cam = monk_spell_t::composite_aoe_multiplier( state );
+
+    if ( p()->buff.unity_within->check() )
+      cam *= 1 + p()->talent.conduit_of_the_celestials.unity_within_dmg_mult->effectN( 1 ).percent();
+
+    return cam;
+  }
```

Comment:
Not dependent on state, so a different override would be more appropriate.

Can be autoparsed-ish. Refer to implementation of `scalding_brew` autoparsing in `keg_smash_t`.